test/cocotb/osdtestlib/eval_utility.py

```diff
@@ -54,9 +59,123 @@ def evaluate_response(self, dut, response, destination, source, reg_width,
         if result:
             # test was successful if none of the above mentioned errors occurred
             dut._log.info("Passed the test!")
+            dut._log.info("")
 
             # clear the buffer for the next read_register()
             del response[:]
 
-        for _ in range(1):
-            yield RisingEdge(dut.clk)
+
+        yield RisingEdge(dut.clk)
+
+
+    @cocotb.coroutine
+    def evaluate_dbpacket(self, dut, standard, id, trace_id, trace_value,
+                          response):
+        """Compares a received debug event packet with its expectation values
+           according to the trace ID and the trace value used to generate the
+           packet
+        Args:
+            dut                device under test.
+            standard           enum variable used to indicate if the packet
+                               should be evaluated according to old or the new
+                               format.
+            id                 ID of the debug module generating the debug event
+                               packet.
+            trace_id           ID of the trace event.
+            trace_value        Value of the trace event.
+            response           read debug event packet.
+        Returns:
+            -
+        """
+
+        # Currently the packets are still generated using the old format
+        # Changing the 'standard' variable will later on also enable the
+        # evaluation according to the new format
+        if (standard == Standard.old):
+
+            # Check the 1st flit
+            if (response[0] != 0x0000):
```

Comment:
Please don't manually check the raw data -- that's what we have introduced the DiPacket class for. This way you also don't have trouble with the changed header format -- it's simply swapping out the logic in the DiPacket class. if you don't have a way to transform raw packet data into DiPacket yet, add a DiPacket.from_raw() method, which parses the packet and sets the attributes.

- Make sure `response` is a `DiPacket`
- Then access each field like if (response.src != xxx) raise ...

Another (possibly even better) option would be to use the evaluate_packet() function that you've written, and extend it with an optional mask argument, telling the function which flits of a packet should be ignored (i.e. the timestamp flit in your case).